File: app/src/main/java/zone/com/zrefreshlayoutdemo/WebViewActivity.java
package zone.com.zrefreshlayoutdemo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.Toast;

import butterknife.BindView;
import butterknife.ButterKnife;
import zone.com.zrefreshlayout.ZRefreshLayout;

/**
 * Created by fuzhipeng on 2017/1/10.
 */

public class WebViewActivity extends AppCompatActivity {

    @BindView(R.id.rotate_header_web_view)
    WebView mWebView;
    @BindView(R.id.refresh)
    ZRefreshLayout refresh;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.web);
        ButterKnife.bind(this);
        mWebView.getSettings().setJavaScriptEnabled(true);
        mWebView.setWebViewClient(new WebViewClient() {
            @Override
            public void onPageFinished(WebView view, String url) {
                refresh.refreshComplete();
            }
        });
        refresh.setPullListener(new ZRefreshLayout.PullListener() {
            @Override
            public void refresh(final ZRefreshLayout zRefreshLayout) {
                mWebView.loadUrl("http://luhaoaimama1.github.io");
            }

        });

        refresh.setLoadMoreListener(new ZRefreshLayout.LoadMoreListener() {
            @Override
            public void loadMore(final ZRefreshLayout zRefreshLayout) {
                refresh.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        Toast.makeText(WebViewActivity.this, "load", Toast.LENGTH_SHORT).show();
                        refresh.loadMoreComplete();
                    }
                }, 2000);

            }

        });
    }
}

File: app/src/main/java/zone/com/zrefreshlayoutdemo/WebViewActivity.kt
package zone.com.zrefreshlayoutdemo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.webkit.WebView
import android.webkit.WebViewClient
import android.widget.Toast
import kotlinx.android.synthetic.main.web.*
import zone.com.zrefreshlayout.ZRefreshLayout

/**
 * Created by fuzhipeng on 2017/1/10.
 */

class WebViewActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.web)
        rotate_header_web_view.settings.javaScriptEnabled = true
        rotate_header_web_view.webViewClient = object : WebViewClient() {
            override fun onPageFinished(view: WebView, url: String) {
                refresh.refreshComplete()
            }
        }
        refresh.pullListener = ZRefreshLayout.PullListener { rotate_header_web_view.loadUrl("http://luhaoaimama1.github.io") }

        refresh.loadMoreListener = ZRefreshLayout.LoadMoreListener {
            refresh.postDelayed({
                Toast.makeText(this@WebViewActivity, "load", Toast.LENGTH_SHORT).show()
                refresh.loadMoreComplete()
            }, 2000)
        }
    }
}
